### PR TITLE
Add mod reinstall dialog

### LINKED ISSUE #1354: Ability to Downgrade/Change Version of Installed Mods
### Role

I am a modpack developer.

### Suggestion

The ability to change an installed mod's version to any version(including older versions) without deleting and reinstalling the mod.

### Benefit

This would save time whenever a mod update crashes the game, as you'd only need to click one or two buttons instead of deleting the mod then installing it again.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

ATLauncher has a similar feature to this, where you can select a mod and "reinstall" it, which lets you select whatever mod version you want.

### PR DESCRIPTION
**Pull request by Trial97**
fixes #1354

Added one more button to the mods page that will open a dialog with the selected mod allowing you to select a new version to install.
Also added a tag to the installed version combobox to determine what is the installed version.

### CODE REVIEW

File: launcher/ui/pages/modplatform/ModPage.cpp [hunk 1]
```diff
@@ -129,13 +131,19 @@ void ModPage::updateVersionList()
...
+        QString flag;
+        if (installedIndex == -1 && installedVersion.isValid() && installedVersion == version.fileId) {
+            flag = tr("[installed]");
+            installedIndex = i;
+        }
         // Only add the version if it's valid or using the 'Any' filter, but never if the version is opted out
         if ((valid || m_filter->versions.empty()) && !optedOut(version)) {
             auto release_type = version.version_type.isValid() ? QString(" [%1]").arg(version.version_type.toString()) : "";
-            m_ui->versionSelectionBox->addItem(QString("%1%2").arg(version.version, release_type), QVariant(i));
+            m_ui->versionSelectionBox->addItem(QString("%1%2%3").arg(version.version, release_type, flag), QVariant(i));
```
Comment (getchoo) [inline launcher/ui/pages/modplatform/ModPage.cpp]:
```suggestion
            m_ui->versionSelectionBox->addItem(QString("%1%2 %3").arg(version.version, release_type, flag), QVariant(i));
```
the space could also go in the original variable, but things like `[release][installed]` is kinda ugly to me

**Review by getchoo — CHANGES_REQUESTED** [review]
i'd recommend the button's name be changed to "change version" similar to the in version tab of instance settings - since that's a bit more descriptive, while "reinstall mod" sounds more like a troubleshooting method in the case a mod was installed incorrectly. it would also probably be nicer to have it near "check for updates" because of their similar functions

File: launcher/ui/pages/modplatform/ModPage.cpp (hunk 1 above)
Comment (Trial97) [reply launcher/ui/pages/modplatform/ModPage.cpp]: I understand that the current flags are ugly, but I have negative UX skills so if you make any kind of suggestions please be very specific. For now added the space before it and also moved and renamed the button.

**Review by TayouVR — APPROVED** [review]
works 👍

**Review by Ryex — APPROVED** [review]

**Comment by Trial97** [comment]
@getchoo other than java this is the last PR that you need to explicitly pay attention to.
When you have time please have a look at this also.

**Review by getchoo — APPROVED** [review]
code looks good and this works as expected

i did notice something during testing though: when modpacks are installed, it seems no metadata is fetched along with it. this in turn causes this feature to be unavailable until the user does something to cause metadata fetching manually like checking for updates. this is obviously out of scope for this PR, but probably something to come back to in the future

**Comment by Trial97** [comment]
That is already tracked in this issue:https://github.com/PrismLauncher/PrismLauncher/issues/1730

**Pull request merged**